apps/dynamo/lib/dynamo/vnode.ex

```diff
@@ -81,6 +105,35 @@ defmodule Vnode do
     end
   end
 
+  defp wait_read_response(key, value, index, context, state, current_read, num_read) do
+    if current_read < num_read do
+      receive do
+        {:ok, ^key, {other_value, other_context}, other_index} ->
```

Comment:
Where is the base case which returns the same key, value, context? and catch all case?

Reply:
We dont need to have same key, value and context, just same key everything else can be checked by comparing the contexts.
I'll write a case for all other cases